integration_tests/src/utils.rs

```diff
@@ -261,3 +324,370 @@ pub fn set_sigterm() {
+        info!("Starting inner thraed");
+        // set the host of this thread to the ns
+        let nsfd = get_nsfd(node.get_name());
+        setns(nsfd, CloneFlags::CLONE_NEWNET).expect("Couldn't set network namespace");
+        let runner = System::new();
+        runner.block_on(async move {
+            let client = awc::Client::default();
+            let mut req = client
+                .get("http://localhost:4877/debts".to_string())
+                .send()
+                .await
+                .expect("Failed to make request to rita RPC");
+            let res = match req.json().await {
+                Err(e) => panic!("Why is get debts failing: {}", e),
+                Ok(a) => a,
+            };
+            *response.write().unwrap() = Some(res);
+        })
+    });
+
+    // wait for the child thread to finish performing it's query
+    while response_local.read().unwrap().is_none() {
+        info!("Waiting for a rpc response from {}", node_name);
+        sleep(Duration::from_millis(100)).await;
+    }
+    sleep(Duration::from_millis(100)).await;
+    let list = response_local.read().unwrap().clone().unwrap();
+
+    let mut ret = None;
+    for e in list {
+        if e.identity.mesh_ip.to_string() == for_node_ip {
+            ret = Some(e.clone());
+            info!("Relevant node debt is {:?}", e);
+        }
+    }
+    ret.unwrap()
+}
+
+#[derive(Debug, Clone)]
+pub struct ValidatorKeys {
+    /// The validator key used by this validator to actually sign and produce blocks
+    pub validator_key: CosmosPrivateKey,
+    // The mnemonic phrase used to generate validator_key
+    pub validator_phrase: String,
+}
+// Simple arguments to create a proposal with
+pub struct RegisterCoinProposalParams {
+    pub coin_metadata: Metadata,
+
+    pub proposal_title: String,
+    pub proposal_desc: String,
+}
+
+pub fn get_deposit() -> Coin {
+    Coin {
+        denom: STAKING_TOKEN.to_string(),
+        amount: 1_000_000_000u64.into(),
+    }
+}
+
+pub fn get_fee(denom: Option<String>) -> Coin {
+    match denom {
+        None => Coin {
+            denom: STAKING_TOKEN.to_string(),
+            amount: MIN_GLOBAL_FEE_AMOUNT.into(),
+        },
+        Some(denom) => Coin {
+            denom,
+            amount: MIN_GLOBAL_FEE_AMOUNT.into(),
+        },
+    }
+}
+
+pub async fn vote_yes_with_retry(
+    contact: &Contact,
+    proposal_id: u64,
+    key: impl PrivateKey,
+    timeout: Duration,
+) {
+    const MAX_VOTES: u64 = 5;
+    let mut counter = 0;
+    let mut res = contact
+        .vote_on_gov_proposal(
+            proposal_id,
+            VoteOption::Yes,
+            get_fee(None),
+            key.clone(),
+            Some(timeout),
+        )
+        .await;
+    while let Err(e) = res {
+        contact.wait_for_next_block(TOTAL_TIMEOUT).await.unwrap();
+        res = contact
+            .vote_on_gov_proposal(
+                proposal_id,
+                VoteOption::Yes,
```

Comment:
why are these failing? timeout?

Reply:
Not sure what you mean here, doesnt seem like its failing?

Reply:
I mean why do you need the retry?